phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java

```diff
@@ -957,4 +962,35 @@ public static int getClientVersion(Scan scan) {
     public static void setClientVersion(Scan scan, int version) {
         scan.setAttribute(BaseScannerRegionObserver.CLIENT_VERSION, Bytes.toBytes(version));
     }
+
+    public static void getDummyResult(byte[] rowKey, List<Cell> result) {
+        result.clear();
```

Comment:
I didn't understand this part. Shouldn't `result` always be empty when it reaches this point? And if it isn't, aren't we overwriting actually the read data?

Reply:
Yes, it should be always empty. I will remove the clear() call.